libraries/standard/http/cbmc/sources/http_cbmc_state.c

```diff
@@ -73,3 +73,35 @@ bool isValidHttpRequestInfo( const HTTPRequestInfo_t * pRequestInfo )
 
     return isValid;
 }
+
+HTTPResponse_t * allocateHttpResponse( HTTPResponse_t * pResponse )
+{
+    if( pResponse == NULL )
+    {
+        pResponse = mallocCanFail( sizeof( HTTPResponse_t ) );
+    }
+
+    if( pResponse != NULL )
+    {
+        __CPROVER_assume( pResponse->bufferLen < CBMC_MAX_OBJECT_SIZE );
+        pResponse->pBuffer = mallocCanFail( pResponse->bufferLen );
+        __CPROVER_assume( pResponse->bodyLen < CBMC_MAX_OBJECT_SIZE );
+        pResponse->pBody = mallocCanFail( pResponse->bodyLen );
+    }
+
+    return pResponse;
+}
+
+bool isValidHttpResponse( const HTTPResponse_t * pResponse )
+{
+    bool isValid = true;
+
+    if( pResponse )
+    {
+        isValid = pResponse->bufferLen < CBMC_MAX_OBJECT_SIZE &&
+                  pResponse->bodyLen < CBMC_MAX_OBJECT_SIZE &&
+                  pResponse->pHeaderParsingCallback == NULL;
```

Comment:
This Callback==NULL is a surprise to see.  Probably want a line to comment to explain why this is reasonable.

Reply:
Oh thanks for noticing. This was left here by mistake when I was trying to debug something.

Reply:
If it is a mistake, should it be removed?

Reply:
It has been removed.